com.dynamo.cr/com.dynamo.cr.bob.test/src/com/dynamo/bob/pipeline/GuiBuilderTest.java

```diff
@@ -89,8 +94,108 @@ private void registerProtoBuilderNames() {
+        src.append("}\n");
+    }
+
+    private void addTemplateNode(StringBuilder src, String id, String parent, String template) {
+        src.append("nodes {\n");
+        src.append("  type: TYPE_TEMPLATE\n");
+        src.append("  id: \""+id+"\"\n");
+        src.append("  parent: \""+parent+"\"\n");
+        src.append("  template: \""+template+"\"\n");
+        src.append("}\n");
+    }
+
+    private void startOverridedNode(StringBuilder src, String type, String id, String parent, List<Integer> overrides) {
+        src.append("nodes {\n");
+        src.append("  type: "+type+"\n");
+        src.append("  id: \""+id+"\"\n");
+        src.append("  parent: \""+parent+"\"\n");
+        for(int num : overrides) {
+            src.append("  overridden_fields: "+num+"\n");
+        }
+    }
+
+    private void finishOverridedNode(StringBuilder src) {
+        src.append("}\n");
+    }
+
+    private void startLayout(StringBuilder src, String name) {
+        src.append("layouts {\n");
+        src.append("  name: \""+name+"\"\n");
+    }
+
+    private void finishLayout(StringBuilder src) {
+        src.append("}\n");
+    }
+
     @Test
     public void test() throws Exception {
         // Kept empty as a future working template
     }
+
+    // https://github.com/defold/defold/issues/6151
+    @Test
+    public void testDefaultLayoutOverridesPriorityOverTemplateValues() throws Exception {
+        StringBuilder src = createGui();
+        addBoxNode(src, "box", "");
+        addTextNode(src, "text", "box", "templateText");
+        addFile("/template.gui", src.toString());
+
+        src = createGui();
+        addBoxNode(src, "box", "");
+        addTemplateNode(src, "template", "box", "/template.gui");
+
+        // override text in default layout
+        startOverridedNode(src, "TYPE_TEXT", "template/text", "template/box", Arrays.asList(8));
+        src.append("  text: \"defaultText\"\n");
+        finishOverridedNode(src);
+
+        startLayout(src, "Landscape");
+        addBoxNode(src, "template/box", "");
+
+        // override clipping_visible in Landscape layout
+        startOverridedNode(src, "TYPE_TEXT", "template/text", "template/box", Arrays.asList(28));
+        src.append("  clipping_visible: false\n");
+        src.append("  text: \"defaultText\"\n");
+        finishOverridedNode(src);
+
+        finishLayout(src);
+        Gui.SceneDesc gui = buildGui(src, "/test.gui");
+        for(LayoutDesc layout : gui.getLayoutsList()) {
+            if (layout.getName().equals("Landscape")) {
+                for(NodeDesc node : layout.getNodesList()) {
+                    if (node.getId().equals("template/text")) {
+                        Assert.assertFalse(node.getClippingVisible());
+                        Assert.assertEquals(node.getText(), "defaultText");
+                        return;
+                    }
+                }
+            }
+        }
+        Assert.assertFalse("Can't find node!", true);
```

Comment:
I like this test a lot.
However I think a last good improvement here is to put this test into a function as well?

Reply:
Thx. Done!